design/api.md

```diff
@@ -11,91 +11,85 @@ This spec describes
 
 Today we use a REST API. In the future we may switch to an RPC-based design. When/if this happens mostly depends on a plan for implementing the UCAN invocation spec.
 
+| Method | Path | Comment |
+|--------|------|---------|
+| POST | [`/api/v0/auth/email/verify`](#post-apiv0authemailverify) | Trigger email verification |
+| POST | [`/api/v0/account`](#post-apiv0account) | Create a full account |
+| GET | [`/api/v0/account/:did`](#get-apiv0accountdid) | Get account information by DID |
+| GET | [`/api/v0/account/:username/did`](#get-apiv0accountusernamedid) | Get the did of an account by username |
+| PUT | [`/api/v0/account/:did/username/:username`](#put-apiv0accountdidusernameusername) | Change an account's username |
+| GET | [`/api/v0/capabilities`](#get-apiv0capabilities) | Get capabilities for a given account |
```

Comment:
> Get capabilities for a given account

Which is in the auth header I guess? Why not

```
GET /api/v0/account/:did/capabilities/
```

and accept some proof that they're the root user, or a UCAN dlegeation chain that proves that they're able to do stuff with the account? Looking below this may be a case where we're bleeding bewteen REST and RPC a bit, which is fine, but worth noting that we're kind of straddling the two design philosophies.

Reply:
> and accept some proof that they're the root user, or a UCAN delegation chain that proves that they're able to do stuff with the account?

I guess that'd work, but just be less general.
IMO it could totally be account agnostic.

Another thing this enabled would be e.g. if you're ssh-ing into the server, and have the fission-cli installed, you may be able to just do stuff by using the server's DID.
The server doesn't really have a fission account. But the main server UCAN would be a valid authorization!

Reply:
> I guess that'd work, but just be less general.

REST vs RPC summed up in one sentence 😛

Reply:
Hm, in this case I'd argue it's more than just RPC vs. REST, it's also that it'd be confusing if we'd use `/account/` in the path segment, since the DID doesn't need to belong to an account.

Reply:
hehe true, but the joke is more that REST is about the single, complete resource (typically a database row), so it's aimed more at simplicity at the sacrifice of efficiency and flexibility

Reply:
Yeah, there's not really a simple DB row for this endpoint.
I guess a `GET /api/v0/did/:did/capabilities/` would be appropriate? Or perhaps just `GET /api/v0/capabilities/:did`?
Although IIUC this is mostly bike-shedding, right? Duplicating the DID from the auth token into the path doesn't actually change how it works/security, does it?

Reply:
yeah it doesn't really matter, and these will likely turn into some invocation-based RPC at some stage